src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.models import RepositoryProjectPathConfig
+from sentry.api.serializers import serialize
+
+
+def get_link(config, filepath, version):
+    oi = config.organization_integration
+    integration = oi.integration
+    install = integration.get_installation(oi.organization_id)
+
+    formatted_path = filepath.replace(config.stack_root, config.source_root)
+
+    return install.get_stacktrace_link(config.repository, formatted_path, version)
+
+
+class ProjectStacktraceLinkEndpoint(ProjectEndpoint):
+    """
+        Returns valid links for source code providers so that
+        users can go from the file in the stack trace to the
+        provider of their choice.
+
+        `filepath`: The file path from the strack trace
+        `commitId` (optional): The commit_id for the last commit of the
+                               release associated to the stack trace's event
+
+    """
+
+    def get(self, request, project):
+        # should probably feature gate
+        filepath = request.GET.get("file")
+        if not filepath:
+            return Response("Filepath is required", status=400)
+
+        commitId = request.GET.get("commitId")
+        result = {"config": None}
+
+        configs = RepositoryProjectPathConfig.objects.filter(project=project)
+        # we only want to attempt the stack trace linking IF the user
+        # has a configuration set up, otherwise we don't care
+        if not configs:
+            return Response(result)
+
+        for config in configs:
+            if not filepath.startswith(config.stack_root):
+                continue
+
+            version = commitId or config.default_branch
```

Comment:
@MeredithAnya should we have an error if `version` is missing since `default_branch` is optional?

Reply:
hmm well I think the error will be when we make the request, since that will be provider specific. right?

Reply:
@MeredithAnya We'll probably get _some_ sort of error depending. I think for now we can leave this alone but I think we might have to revisit this later. Alternatively, we could make `default_branch` required at some point.

Reply:
@scefali I wonder if we can make that a validation error when saving the config, like if you trying to save a config that is for a certain provider, make sure the specify a branch. iono

Reply:
@MeredithAnya yea, that works too. We can put some logic into the updater/creator of the endpoint to check the default branch for certain providers.